lib/cnet/ipv4/ip4_output.c

```diff
@@ -63,24 +65,28 @@ ip4_header(pktmbuf_t *m, uint16_t nxt)
     m->l3_len = sizeof(struct cne_ipv4_hdr);
 
     l4 = pktmbuf_mtod(m, void *);
-    ip = (struct cne_ipv4_hdr *)pktmbuf_adj_offset(m, -m->l3_len);
+    ip = (struct cne_ipv4_hdr *)pktmbuf_prepend(m, m->l3_len);
     if (!ip)
-        return CNE_EDGE_ID_INVALID;
-    ip->dst_addr        = md->faddr.cin_addr.s_addr;
-    ip->src_addr        = md->laddr.cin_addr.s_addr;
-    ip->time_to_live    = pcb->ttl;
-    ip->type_of_service = pcb->tos;
-    ip->next_proto_id   = pcb->ip_proto;
+        return nxt;
+
     ip->version_ihl     = (IPv4_VERSION << 4) | (sizeof(struct cne_ipv4_hdr) / 4);
+    ip->type_of_service = pcb->tos;
     ip->total_length    = htobe16(pktmbuf_data_len(m));
+    ip->packet_id       = 0;
```

Comment:
not necessary since you're setting this to nif->ip_ident later anyway.

Reply:
Removed.